File: frontend/lib/features/challenges/data/models/requests/challenge_participation.dart
class ChallengeParticipationUpdateRequestModel {
  final String color;
  final DateTime startDate;
  final bool notificationsReminderEnabled;
  final String? reminderTime;
  final String? reminderBody;
  final bool finished;

  const ChallengeParticipationUpdateRequestModel({
    required this.color,
    required this.startDate,
    required this.notificationsReminderEnabled,
    required this.reminderTime,
    required this.reminderBody,
    required this.finished,
  });

  Map<String, dynamic> toJson() {
    return {
      'color': color,
      'start_date': startDate.toUtc().toIso8601String(),
      'notifications_reminder_enabled': notificationsReminderEnabled,
      'reminder_time': reminderTime,
      'reminder_body': reminderBody,
      'finished': finished,
    };
  }
}

class ChallengeParticipationCreateRequestModel {
  final String challengeId;
  final String color;
  final DateTime startDate;

  const ChallengeParticipationCreateRequestModel({
    required this.challengeId,
    required this.color,
    required this.startDate,
  });

  Map<String, dynamic> toJson() {
    return {
      'challenge_id': challengeId,
      'color': color,
      'start_date': startDate.toUtc().toIso8601String(),
    };
  }
}

File: backend/src/features/challenges/structs/requests/challenge_participation.rs
use chrono::{DateTime, NaiveTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Deserialize)]
pub struct GetChallengeParticipationParams {
    pub challenge_participation_id: Uuid,
}

#[derive(Deserialize)]
pub struct UpdateChallengeParticipationParams {
    pub challenge_participation_id: Uuid,
}

#[derive(Deserialize, Serialize)]
pub struct ChallengeParticipationUpdateRequest {
    pub color: String,
    pub start_date: DateTime<Utc>,
    pub notifications_reminder_enabled: bool,
    pub reminder_time: Option<NaiveTime>, // UTC
    pub reminder_body: Option<String>,
}

#[derive(Deserialize)]
pub struct ChallengeParticipationCreateRequest {
    pub challenge_id: uuid::Uuid,
    pub color: String,
    pub start_date: DateTime<Utc>,
}
